src/Draco.Compiler/Internal/Binding/Binder_BoundExpression.cs

```diff
@@ -160,9 +194,81 @@ private BoundExpression TypeIndirectCallExpression(UntypedIndirectCallExpression
     {
-        var typedLeft = this.TypeLvalue(assignment.Left, constraints, diagnostics);
         var typedRight = this.TypeExpression(assignment.Right, constraints, diagnostics);
         var compoundOperator = assignment.CompoundOperator?.Result;
+
+        // NOTE: This is how we deal with properties and indexers
+        if (assignment.Left is UntypedPropertySetLvalue prop)
+        {
+            if (prop.Setter.IsError)
+            {
+                return new BoundReferenceErrorExpression(prop.Syntax, prop.Setter);
+            }
+            return new BoundPropertySetExpression(
+                assignment.Syntax,
+                prop.Receiver is null ? null : this.TypeExpression(prop.Receiver, constraints, diagnostics),
+                prop.Setter,
+                compoundOperator is not null
+                    ? this.CompoundPropertyExpression(
+                        assignment.Syntax,
+                        prop.Receiver is null ? null : this.TypeExpression(prop.Receiver, constraints, diagnostics),
+                        typedRight,
+                        ((IPropertyAccessorSymbol)prop.Setter).Property,
+                        compoundOperator,
+                        ImmutableArray<BoundExpression>.Empty,
+                        diagnostics)
+                    : typedRight);
+        }
+
+        if (assignment.Left is UntypedIndexSetLvalue index)
+        {
+            if (index.Setter.Result.IsError)
+            {
+                return new BoundReferenceErrorExpression(index.Syntax, index.Setter.Result);
+            }
+            return new BoundIndexSetExpression(
+                assignment.Syntax,
+                this.TypeExpression(index.Receiver, constraints, diagnostics),
+                index.Setter.Result,
+                compoundOperator is not null
+                    ? this.CompoundPropertyExpression(assignment.Syntax,
+                        this.TypeExpression(index.Receiver, constraints, diagnostics),
+                        typedRight,
+                        ((IPropertyAccessorSymbol)index.Setter.Result).Property,
+                        compoundOperator,
+                        index.Indices.Select(x => this.TypeExpression(x, constraints, diagnostics)).ToImmutableArray(),
+                        diagnostics)
+                    : typedRight,
+                index.Indices.Select(x => this.TypeExpression(x, constraints, diagnostics)).ToImmutableArray());
+        }
+
+        else if (assignment.Left is UntypedMemberLvalue mem && mem.Member.Result[0] is PropertySymbol pr)
+        {
+            var setter = pr.Setter;
+            if (setter is null)
+            {
+                diagnostics.Add(Diagnostic.Create(
+                    template: SymbolResolutionErrors.CannotSetGetOnlyProperty,
+                    location: assignment.Syntax?.Location,
+                    pr.FullName));
+                setter = new NoOverloadFunctionSymbol(1);
+            }
```

Comment:
This seems like logic duplication from the untyped binding, is it because of the member expression can delay what kind of symbol this is?